core/coreunix/add.go

```diff
@@ -470,6 +472,9 @@ func (adder *Adder) addFile(file files.File) error {
 		return err
 	}
 
+	if !strings.EqualFold(adder.WpName, "") && adder.Wrap {
+		return adder.addNode(dagnode, adder.WpName)
```

Comment:
I'd rather avoid duplicating the function call, we can define a new variable here, e.g., `addFileName := file.FileName()`, and if that name is `""` we set it to `adder.Name`. e.g.,

```Go
addFileName := file.FileName()
if addFileName == "" && adder.Name != "" {
  addFileName = adder.Name

  adder.Name = ""
  // Unset the `Name` option to avoid applying the same name to many unnamed files.
  // TODO: This could be turned into a FIFO queue to handle the scenario with more than one unnamed file.
}

adder.addNode(dagnode, addFileName)
```

We do two checks:
1. Apply the `--name` option only to unnamed files (e.g., coming from `stdin`).
2. Apply the `--name` option only once (since I fear that the HTTP commands library may be able to send many unnamed files), otherwise we would have many files with the same name in one directory.